ni_measurementlink_service/discovery/_client.py

```diff
@@ -111,13 +114,15 @@ def register_service(self, service_info: ServiceInfo, service_location: ServiceL
         Returns:
             ID that can be used to unregister the service.
         """
+        annotations = service_info.annotations
```

Comment:
I don't think this is making a copy of `service_info.annotations`. In the debugger, check if service_info.annotations is modified with line 118.
https://stackoverflow.com/questions/2465921/how-to-copy-a-dictionary-and-only-edit-the-copy

Reply:
True